connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -338,6 +339,38 @@ protected Connector getConnector(String connType) {
         }
     }
 
+    /**
+     * Checks a given {@link ConfigInfos} for validation error messages and adds an exception
+     * to the given {@link Callback} if any were found.
+     * @param config ConfigInfos to Read Errors from
+     * @param callback Callback to add Config Error Exception to
+     * @return TRUE iff Errors were Found in the Config
+     */
+    protected final boolean maybeAddConfigErrors(final ConfigInfos config,
```

Comment:
@ewencp I was striving to keep the comments around `final` minimal. By variables above I'm referring to references, unless we are talking about primitive types. And, of course, in this case reassignment is the only mutation that you may apply to them. 

Having in mind the distinction between primitives and references (pointers elsewhere), I didn't imply that `final` prevents you from mutating the object itself (where the reference/pointer points to. For instance a collection). Apologies if this wasn't obvious above (but I was referring to variables, meaning primitives or references, not values).